### PR TITLE
Switch to docker-container provider for buildx

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Now the inline build cache works. Thanks @bendavies for your time.

Next step: Switch away from inline cache so that we can use `mode=max`?

### CODE REVIEW

**Comment by bendavies** [comment]
`--output=type=docker` is just `--load`?

**Comment by teohhanhui** [comment]
`buildx build` has the `--load` shortcut, but since we're using `buildx bake`...

**Comment by bendavies** [comment]
how did that fix the caching? why were the `--cache-from` options not working?

**Comment by teohhanhui** [comment]
In other news, there should be a big warning sign not to use the `docker` provider considering how it doesn't work properly. Lol...

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -176,23 +182,31 @@ jobs:
...
       - name: Install Docker Compose
         run: |
-          docker create --name docker_compose "docker/compose:alpine-v${DOCKER_COMPOSE_VERSION}"
+          docker create --name docker_compose "docker/compose:alpine-${DOCKER_COMPOSE_VERSION}"
           docker cp docker_compose:/usr/local/bin/docker-compose /usr/local/bin/docker-compose
           docker rm docker_compose
+      - name: Create builder instance
+        run: |
+          docker context create default_tls
+          docker buildx create --driver docker-container --driver-opt image=moby/buildkit:master,network=host --use default_tls
```
Comment (bendavies) [inline .github/workflows/ci-cd.yml]: FYI i used `--driver-opt image=moby/buildkit:master` to include https://github.com/moby/buildkit/pull/1382

**Pull request merged**